### PR TITLE
Experts code submission

### PR DESCRIPTION
**Pull request by nazlicantrn**
# Expert Medical CLIP Training & Evaluation Pipeline

## Summary

This PR introduces a comprehensive pipeline for training and evaluating domain-specialized CLIP models on medical imaging tasks (dermatology and ophthalmology). The pipeline includes data processing, GPT-4 caption enhancement with cost tracking, Optuna hyperparameter optimization and multi-faceted evaluation benchmarks.

## What's Included

**Dataset Processing Script for the Experts & Data Splitting**
- Complete preprocessing pipeline for expert medical datasets with multimediset JSONL format standardization and image validation

**New GPT Caption Augmentation Scripts**
- Updated robust caption enhancement with medical context-aware prompts
- Added cost estimation scripts for budget planning before API calls
- Batch processing with error handling and retry logic
- Documented usage

**Hyperparameter Optimization (Optuna) for Disease Classification Tasks**
- Automated search code for learning rate, weight decay and frozen layers
- Study persistence and visualization tools

**Evaluation Benchmarks**
- **Classification**: Frozen CLIP embeddings → MLP classifier with per-class metrics
- **Random Retrieval**: 4-way forced choice (Recall@1 with 25% baseline)
- **Hard Negative Retrieval**: Visually similar but semantically different negatives benchmark
- **Skin Tone Stratified**: Fairness evaluation across Fitzpatrick scale groups
- Diagnostic probes for text/image tower sanity checks
- Qualitative visualization and lexical overlap analysis

**Documentation**
- Comprehensive READMEs for classification and retrieval benchmarks
- Complete `requirements_experts.txt` with usage examples

### CODE REVIEW

File: src/multimeditron/experts/gpt_augmentation_pipeline/collect_all.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+from openai import OpenAI
+from pathlib import Path
+import json, time, re
+from typing import List
+import config
+
+client = OpenAI(api_key=config.OPEN_API_KEY)
```
[the remark this answers is not in the document]
Comment (nazlicantrn) [reply src/multimeditron/experts/gpt_augmentation_pipeline/collect_all.py]: changed

File: src/multimeditron/experts/gpt_augmentation_pipeline/make_batches.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+- Explicitly note if certain structures or details are not visible.
+- Do not use markdown or bullet formatting in the generated description; start directly with prose.
+- Keep tone like an ophthalmologist documenting structured findings for dataset curation, not a patient note.
+
+Now here is the info:
+-----
+"""
+
+SKIN_CONTENT= "You are an expert assistant for dermatology dataset annotation."
+OPT_CONTENT= "You are an expert assistant for opthalmology dataset annotation."
```
[the remark this answers is not in the document]
Comment (nazlicantrn) [reply src/multimeditron/experts/gpt_augmentation_pipeline/make_batches.py]: resolved

File: src/multimeditron/experts/gpt_augmentation_pipeline/utils.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import config
+
+# ---------------------------------------------------------------------
+# Paths
+# ---------------------------------------------------------------------
+
+DATASET_ROOT = Path(config.DATASET_ROOT)
+IMAGE_ROOT = Path(config.DATASET_IMG_ROOT)
+
+MANIFEST_NAME = os.getenv(
```
[the remark this answers is not in the document]
Comment (nazlicantrn) [reply src/multimeditron/experts/gpt_augmentation_pipeline/utils.py]: resolved
